format/ComputeIR.fbs

```diff
@@ -0,0 +1,210 @@
+
+  /// The arguments passed to `relation_name`.
+  arguments: [Relation] (required);
+
+  /// The schema of rows in this Relation
+  schema: Schema;
+}
+
+/// The contents of Relation.options will be FilterOptions
+/// if Relation.name = "filter"
+table FilterOptions {
+  /// The expression which will be evaluated against input rows
+  /// to determine whether they should be excluded from the
+  /// "filter" relation's output.
+  filter_expression: Expression (required);
+}
+
+/// The contents of Relation.options will be ProjectOptions
+/// if Relation.name = "project"
+table ProjectOptions {
+  /// Expressions which will be evaluated to produce to
+  /// the rows of the "project" relation's output.
+  expressions: [Expression] (required);
+}
+
+/// The contents of Relation.options will be AggregateOptions
+/// if Relation.name = "aggregate"
+table AggregateOptions {
+  /// Expressions which will be evaluated to produce to
+  /// the rows of the "aggregate" relation's output.
+  aggregations: [Expression] (required);
+  /// Keys by which `aggregations` will be grouped.
+  keys: [Expression];
+}
+
+enum JoinKind : int {
+  INNER,
+  LEFT,
+  RIGHT,
+  FULL,
+  SEMI,
+  ANTI,
+}
+
+/// The contents of Relation.options will be JoinOptions
+/// if Relation.name = "join"
+table JoinOptions {
+  /// The expression which will be evaluated against rows from each
+  /// input to determine whether they should be included in the
+  /// "join" relation's output.
+  on_expression: Expression (required);
+  join_kind: JoinKind;
+}
+
+/// Whether lesser values should precede greater or vice versa.
+enum Ordering : uint8 {
+  ASCENDING,
+  DESCENDING,
+}
+
+/// Whether nulls should precede or follow other values.
+enum NullOrdering : uint8 {
+  FIRST,
+  LAST
+}
+
+table SortKey {
+  value: Expression (required);
+  ordering: Ordering = ASCENDING;
+  null_ordering: NullOrdering = LAST;
+}
+
+/// The contents of Relation.options will be OrderByOptions
+/// if Relation.name = "order_by"
+table OrderByOptions {
+  /// Define sort order for rows of output.
+  /// Keys with higher precedence are ordered ahead of other keys.
+  keys: [SortKey] (required);
+}
+
+/// The contents of Relation.options will be LimitOptions
+/// if Relation.name = "limit"
+table LimitOptions {
+  /// Set the maximum number of rows of output.
+  count: long;
+}
+
+/// The contents of Relation.options will be LiteralOptions
+/// if Relation.name = "literal"
+table LiteralOptions {
+  /// Batches of rows in this literal.
+  batches: [RecordBatch] (required);
+  /// A dictionary batch for this literal.
+  dictionary_batch: DictionaryBatch;
+}
+
+/// The contents of Relation.options will be InteractiveOutputOptions
+/// if Relation.name = "interactive_output"
+table InteractiveOutputOptions {
+  /// In an interactive context there is an implicit pipe open between
+  /// the producer and the consumer along which output may be sent.
+  /// `id` is an opaque identifier for such outputs.
```

Comment:
```suggestion
  /// `id` is an opaque identifier for such outputs; for example it could be used to
  /// identify which Jupyter output cell should receive the output.
```